<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TableViewWidget</class>
 <widget class="QWidget" name="TableViewWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>924</width>
    <height>680</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_6">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="frame">
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <property name="spacing">
         <number>3</number>
        </property>
        <property name="leftMargin">
         <number>2</number>
        </property>
        <property name="topMargin">
         <number>4</number>
        </property>
        <property name="bottomMargin">
         <number>6</number>
        </property>
        <item>
         <widget class="QToolButton" name="tableSettingsButton">
          <property name="minimumSize">
           <size>
            <width>22</width>
            <height>22</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>22</width>
            <height>22</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Show/Hide table settings</string>
          </property>
          <property name="styleSheet">
           <string>QToolButton {
border: none;
}
QToolButton:checked {
  border: 1px solid lightgray;
}
</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/tulip/gui/icons/16/settings.png</normaloff>:/tulip/gui/icons/16/settings.png</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>38</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">
font: bold; font-size: 22px; color: #A0A0A0A0</string>
          </property>
          <property name="text">
           <string>Nodes</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QFrame" name="tableSettingsFrame">
        <property name="styleSheet">
         <string notr="true">#tableSettingsFrame {
	      padding-top: 5px;
	      }</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <property name="spacing">
          <number>5</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="spacing">
            <number>5</number>
           </property>
           <property name="sizeConstraint">
            <enum>QLayout::SetMinimumSize</enum>
           </property>
           <property name="leftMargin">
            <number>5</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Show</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="eltTypeCombo">
             <property name="toolTip">
              <string>Select row element type</string>
             </property>
             <property name="sizeAdjustPolicy">
              <enum>QComboBox::AdjustToContents</enum>
             </property>
             <property name="frame">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="filteringLabel">
             <property name="toolTip">
              <string>Only show the rows (nodes or edge) whose
the chosen selection property value is 'true'</string>
             </property>
             <property name="text">
              <string>filtered by</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="filteringPropertyCombo">
             <property name="sizePolicy">
              <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>Only show the rows (nodes or edge) whose
the chosen selection property value is 'true'</string>
             </property>
             <property name="sizeAdjustPolicy">
              <enum>QComboBox::AdjustToContents</enum>
             </property>
             <property name="frame">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="Line" name="line">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="zoomLabel">
             <property name="toolTip">
              <string>Select the table zoom level</string>
             </property>
             <property name="text">
              <string>Zoom</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="zoomSpinBox">
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::PlusMinus</enum>
             </property>
             <property name="keyboardTracking">
              <bool>false</bool>
             </property>
             <property name="suffix">
              <string>%</string>
             </property>
             <property name="minimum">
              <number>60</number>
             </property>
             <property name="maximum">
              <number>300</number>
             </property>
             <property name="singleStep">
              <number>20</number>
             </property>
             <property name="value">
              <number>100</number>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Expanding</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="spacing">
            <number>5</number>
           </property>
           <property name="leftMargin">
            <number>5</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="valueColmunMatch">
             <property name="text">
              <string>with value</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="valueMatchCombo">
             <property name="toolTip">
              <string notr="true">Indicates the type of filtering:
- 'matching' for a regular expression
- 'like' for a sql like pattern.</string>
             </property>
             <property name="sizeAdjustPolicy">
              <enum>QComboBox::AdjustToMinimumContentsLength</enum>
             </property>
             <property name="minimumContentsLength">
              <number>8</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="ClearableLineEdit" name="filterEdit">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>170</width>
               <height>24</height>
              </size>
             </property>
             <property name="toolTip">
              <string>Only show rows (nodes or edges) whose
the chosen column value matches the given regular expression.
Press 'Return' to validate.</string>
             </property>
             <property name="placeholderText">
              <string>a regular expression</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>in</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="matchPropertyButton">
             <property name="toolTip">
              <string>Only show the rows (nodes or edges) whose
the chosen column value matches the given regular expression</string>
             </property>
             <property name="layoutDirection">
              <enum>Qt::RightToLeft</enum>
             </property>
             <property name="text">
              <string>Any</string>
             </property>
             <property name="icon">
              <iconset>
               <normaloff>:/tulip/gui/ui/combobox_arrow.png</normaloff>:/tulip/gui/ui/combobox_arrow.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="toolTip">
              <string>Only show the rows (nodes or edges) whose
the chosen column value contains the given pattern</string>
             </property>
             <property name="text">
              <string>column</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="filtercase">
             <property name="font">
              <font>
               <pointsize>8</pointsize>
               <weight>50</weight>
               <italic>false</italic>
               <bold>false</bold>
              </font>
             </property>
             <property name="toolTip">
              <string>indicates if the match is case sensitive or not</string>
             </property>
             <property name="text">
              <string>case
sensitive</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <property name="spacing">
            <number>5</number>
           </property>
           <property name="leftMargin">
            <number>5</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="labelColumnMatch">
             <property name="toolTip">
              <string>Only show the columns (properties) whose name
matches the given regular expression.</string>
             </property>
             <property name="text">
              <string>Show columns with name</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="columnMatchCombo">
             <property name="toolTip">
              <string notr="true">Indicates the type of filtering
- 'matching' for a regular expressio
- 'like' for a sql like pattern.</string>
             </property>
             <property name="minimumContentsLength">
              <number>8</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="ClearableLineEdit" name="columnsFilterEdit">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>170</width>
               <height>24</height>
              </size>
             </property>
             <property name="toolTip">
              <string>Only show the columns (properties) whose name
matches the given regular expression.
Press 'Return' to validate.</string>
             </property>
             <property name="placeholderText">
              <string>a regular expression</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="columnsfiltercase">
             <property name="font">
              <font>
               <pointsize>8</pointsize>
              </font>
             </property>
             <property name="toolTip">
              <string>indicates if the match is case sensitive or not</string>
             </property>
             <property name="text">
              <string>case
sensitive</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
	<widget class="QFrame" name="frame_2">
	  <property name="styleSheet">
	    <string notr="true">
#frame_2 {
border-right: 0px;
}
QFrame[section] {
border: 1px solid #D9D9D9;
padding-top: 0px;
}</string>
	  </property>
	  <property name="section" stdset="0">
	    <bool>true</bool>
	  </property>
	  <layout class="QVBoxLayout" name="verticalLayout_2">
	    <property name="spacing">
	      <number>0</number>
	    </property>
	    <property name="margin">
	      <number>0</number>
	    </property>
	    <item>
	      <widget class="NavigableTableView" name="table">
		<property name="contextMenuPolicy">
		  <enum>Qt::CustomContextMenu</enum>
		</property>
		<property name="frameShape">
		  <enum>QFrame::NoFrame</enum>
		</property>
		<property name="selectionBehavior">
		  <enum>QAbstractItemView::SelectRows</enum>
		</property>
		<property name="sortingEnabled">
		  <bool>true</bool>
		</property>
		<property name="wordWrap">
		  <bool>false</bool>
		</property>
		<property name="cornerButtonEnabled">
		  <bool>false</bool>
		</property>
		<attribute name="verticalHeaderMinimumSectionSize">
		  <number>30</number>
		</attribute>
	      </widget>
	    </item>
	  </layout>
	</widget>
      </item>
    </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ClearableLineEdit</class>
   <extends>QLineEdit</extends>
   <header location="global">tulip/ClearableLineEdit.h</header>
  </customwidget>
  <customwidget>
   <class>NavigableTableView</class>
   <extends>QTableView</extends>
   <header>navigabletableview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <slots>
  <slot>setPropertyVisible(tlp::PropertyInterface*,bool)</slot>
  <slot>setFilter(QString)</slot>
 </slots>
</ui>
